src/NHibernate/Dialect/Schema/PostgreSQLMetadata.cs

```diff
@@ -138,7 +138,43 @@ public PostgreSQLColumnMetadata(DataRow rs)
 			this.SetNumericalPrecision(rs["NUMERIC_PRECISION"]);
 
 			Nullable = Convert.ToString(rs["IS_NULLABLE"]);
-			TypeName = Convert.ToString(rs["DATA_TYPE"]);
+			TypeName = Normalize(Convert.ToString(rs["DATA_TYPE"]));
+		}
+
+		private static string Normalize(string typeName)
+		{
+			switch (typeName)
+			{
+				case "double precision":
+					return "float8";
+				case "real":
+					return "float4";
+				case "smallint":
+					return "int2";
+				case "integer":
+					return "int4";
+				case "bigint":
+					return "int8";
+			}
+
+			if (typeName.StartsWith("character"))
```

Comment:
Please use `StringComparison.Ordinal` with `StartsWith` and `EndsWith`

Reply:
Done